.github/workflows/backstop.yml

```diff
@@ -0,0 +1,103 @@
+
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ROOT_PASSWORD: root
+        ports:
+        - 3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+    steps:
+      - name: Config mysql database as per CiviCRM requirement
+        run: echo "SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));" | mysql -u root --password=root --host=mysql
+
+      - name: Config amp
+        run : |
+          amp config:set --db_type=mysql_dsn --mysql_dsn='mysql://root:root@mysql:3306' --httpd_type=apache24 --httpd_restart_command='sudo /usr/sbin/apache2ctl graceful' --perm_type=worldWritable --hosts_type=file
+          echo "IncludeOptional /github/home/.amp/apache.d/*.conf" >> /etc/apache2/apache2.conf
+          /usr/sbin/apache2ctl restart
+
+      - name: Build Drupal site
+        run: |
+          civibuild create drupal-clean --civi-ver 5.24.6 --web-root $GITHUB_WORKSPACE/site
+          chmod -R 777 /__w/uk.co.compucorp.civicase/uk.co.compucorp.civicase
+
+      - name: Installing CiviCase and Shoreditch
+        working-directory: ${{ env.CIVICRM_EXTENSIONS_DIR }}
+        run: |
+          git clone https://github.com/compucorp/uk.co.compucorp.civicase.git
+          git clone --depth 1 https://github.com/civicrm/org.civicrm.shoreditch.git
+          cv en shoreditch civicase
+          drush en civicrmtheme -y
+          drush en bootstrap -y
+          drush vset theme_default bootstrap
+
+      - name: Installing Shoreditch Companion Theme
+        working-directory: ${{ env.DRUPAL_THEME_DIR }}
+        run: |
+          git clone https://github.com/compucorp/shoreditch-companion-d7-theme.git depth=1
+          drush en -y shoreditch_companion_d7_theme
+          drush vset civicrmtheme_theme_admin shoreditch_companion_d7_theme
+
+          drush sql-query "UPDATE block SET status = 0 WHERE theme='shoreditch_companion_d7_theme' AND module='civicrm' AND delta IN ('1', '2', '3', '4', '5');" -y
+          drush sql-query "UPDATE block SET status = 0 WHERE theme='shoreditch_companion_d7_theme' AND module='search' AND delta='form';" -y
+          drush sql-query "UPDATE block SET status = 0 WHERE theme='shoreditch_companion_d7_theme' AND module='system' AND delta IN ('navigation', 'powered-by');" -y
+
+          drush cc all && drush cc civicrm
+
+      - name: Setup data and site-config.json for Backstop tests
+        working-directory: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.civicase
+        run: |
+          git checkout "${{ github.head_ref }}"
+          cd tests/backstop_data
+          cp site-config.json.sample site-config.json
+          sed -i 's+<url>+http://localhost:7979+' site-config.json
+          sed -i 's+<path-to-site-root>+/__w/uk.co.compucorp.civicase/uk.co.compucorp.civicase/site/web+' site-config.json
+          npm install
+          npx gulp backstopjs:setup-data
```

Comment:
So in the "Installing CiviCase and Shoreditch" step, you install the extension from a clone pointing to master, but here you setup the data using the scripts from the head branch, which might be a branch that is not compatible with master anymore (e.g, has a different data structure than master) and this can cause issues in the future.
I think that what would make more sense here is to, in the installation step, checkout the head branch before installing the extension